File: infra/stacks/engebreth_website_stack.py
from aws_cdk import CfnOutput, Stack
from constructs import Construct

from infra.constructs import DnsConstruct, StaticSite, VisitorCounterApi


class EngebrethWebsiteStack(Stack):
    def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
        super().__init__(scope, construct_id, **kwargs)

        config = self.node.try_get_context("config")
        domain_name = config["dns"]["fqdn"]

        dns = DnsConstruct(self, "Dns", domain_name=domain_name)

        site = StaticSite(
            self,
            "Site",
            domain_name=domain_name,
            hosted_zone=dns.hosted_zone,
            certificate=dns.certificate,
        )

        api = VisitorCounterApi(self, "Api")

        CfnOutput(self, "ApiUrl", value=api.api.url)
        CfnOutput(self, "DistributionDomainName", value=site.distribution.distribution_domain_name)

File: lib/stacks/engebreth-website-stack.ts
import * as cdk from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { DnsConstruct } from '../constructs/dns';
import { StaticSite } from '../constructs/site';
import { VisitorCounterApi } from '../constructs/api';

export class EngebrethWebsiteStack extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    const config = this.node.tryGetContext('config') as Record<string, any>;
    const domainName = config.dns.fqdn as string;

    const dns = new DnsConstruct(this, 'Dns', { domainName });

    const site = new StaticSite(this, 'Site', {
      domainName,
      hostedZone: dns.hostedZone,
      certificate: dns.certificate,
    });

    const api = new VisitorCounterApi(this, 'Api');

    new cdk.CfnOutput(this, 'ApiUrl', { value: api.api.url });
    new cdk.CfnOutput(this, 'DistributionDomainName', { value: site.distribution.distributionDomainName });
  }
}
